src/query/execute_statement.c

```diff
@@ -17246,6 +17262,651 @@ do_set_timezone (PARSER_CONTEXT * parser, PT_NODE * statement)
+    {
+      assert (statement->info.create_synonym.synonym_name->node_type == PT_NAME);
+
+      synonym_name = statement->info.create_synonym.synonym_name->info.name.original;
+      sm_downcase_name (synonym_name, synonym_downcase_name, SM_MAX_IDENTIFIER_LENGTH);
+
+      /* cannot use catalog class names as synonym names. */
+      if (db_is_system_class_by_name (synonym_downcase_name))
+	{
+	  error = ER_QPROC_CANNOT_USE_CATALOG_NAME;
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 0);
+	  goto end;
+	}
+    }
+
+  /* synonym_owner */
+  if (statement->info.create_synonym.synonym_owner_name)
+    {
+      assert (statement->info.create_synonym.synonym_owner_name->node_type == PT_NAME);
+
+      synonym_owner_name = statement->info.create_synonym.synonym_owner_name->info.name.original;
+      synonym_owner = au_find_user (synonym_owner_name);
+    }
+
+  /* is_public_synonym */
+  if (statement->info.create_synonym.access_modifier == PT_PUBLIC)
+    {
+      /* cannot create public synonym for user. */
+      if (synonym_owner)
+	{
+	  error = ER_QPROC_CANNOT_EXIST_PUBLIC_SYNONYM;
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, au_get_user_name (synonym_owner));
+	  goto end;
+	}
+
+      /* only DBA users and members of the DBA group can create public synonyms. */
+      if (is_dba_group_member)
+	{
+	  synonym_owner = au_get_public_user ();
+	  if (synonym_owner == NULL)
+	    {
+	      assert (er_errid () != NO_ERROR);
+	      error = er_errid ();
+	      goto end;
+	    }
+
+	  is_public_synonym = 1;
+	}
+      else
+	{
+	  error = ER_AU_DBA_ONLY;
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, "CREATE PUBLIC SYNONYM");
+	  goto end;
+	}
+    }
+  else
+    {
+      assert (statement && statement->info.create_synonym.access_modifier == PT_PRIVATE);
+
+      if (synonym_owner == NULL)
+	{
+	  synonym_owner = current_user;
+	}
+
+      /* creating private synonym for other user can only be performed by DBA or a DBA group member. */
+      if (ws_is_same_object (current_user, synonym_owner) == false)
+	{
+	  error = ER_QPROC_CANNOT_PRIAVTE_SYNONYM_FOR_USER;
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 0);
+	  goto end;
+	}
+
+      is_public_synonym = 0;
+    }
+
+  or_replace = statement->info.create_synonym.or_replace;
+
+  /* check if the same synonym exists. */
+  synonym_owner_name = au_get_user_name (synonym_owner);
+  if (do_get_synonym (synonym_downcase_name, synonym_owner_name, is_public_synonym))
+    {
+      if (or_replace)
+	{
+	  has_already = true;
+	}
+      else
+	{
+	  if (is_public_synonym)
+	    {
+	      error = ER_QPROC_PUBLIC_SYNONYM_ALREADY_EXIST;
+	      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, error, 2, synonym_owner_name, synonym_downcase_name);
+	      goto end;
+	    }
+	  else
+	    {
+	      error = ER_QPROC_PRIVATE_SYNONYM_ALREADY_EXIST;
+	      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, error, 2, synonym_owner_name, synonym_downcase_name);
+	      goto end;
+	    }
```

Comment:
suggested shorten version. The decision is yours.

```suggestion
	  error = is_public_synonym ? ER_QPROC_PUBLIC_SYNONYM_ALREADY_EXIST : ER_QPROC_PRIVATE_SYNONYM_ALREADY_EXIST;
	  er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, error, 2, synonym_owner_name, synonym_downcase_name);
	  goto end;
```

Reply:
This is the part i agreed to change in the previous comment.

I will change it